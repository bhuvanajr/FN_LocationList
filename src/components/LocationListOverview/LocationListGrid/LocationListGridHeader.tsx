import React from 'react';

export const LocationListGridHeader: React.FC = () => {
  return (
    <thead className="table-light">
      <tr>
        <th scope="col">Location</th>
        <th scope="col">Location No</th>
        <th scope="col">Chargers</th>
        <th scope="col">Country</th>
        <th scope="col">Last updated</th>
        <th scope="col"></th>
      </tr>
    </thead>
  );
};
