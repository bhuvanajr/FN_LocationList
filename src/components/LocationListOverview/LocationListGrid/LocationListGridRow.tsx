import React from 'react';
import { Location } from 'types';

type Props = {
  data: Location;
};

export const LocationListGridRow: React.FC<Props> = ({ data }) => {
  return (
    <tr>
      <th scope="row" className="text-primary">
        {data.name}
      </th>
      <td>{data.location}</td>
      <td>{data.chargers.length}</td>
      <td>{data.country}</td>
      <td>{data.lastUpdated}</td>
      <td>
        <button type="button" className="btn-icon" onClick={() => {}}>
          <div className="icon icon-edit" />
        </button>
      </td>
    </tr>
  );
};
