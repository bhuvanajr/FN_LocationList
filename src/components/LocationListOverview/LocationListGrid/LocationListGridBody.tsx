import { useLocationListOverviewContext } from 'context';
import React from 'react';
import { LocationListGridRow } from './LocationListGridRow';

export const LocationListGridBody: React.FC = () => {
  const {
    state: { locationList },
  } = useLocationListOverviewContext();

  return (
    <tbody>{locationList && locationList.map((row, index) => <LocationListGridRow data={row} key={index} />)}</tbody>
  );
};
