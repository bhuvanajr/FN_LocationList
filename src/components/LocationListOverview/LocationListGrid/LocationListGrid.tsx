import React from 'react';

import { LocationListGridBody } from './LocationListGridBody';
import { LocationListGridHeader } from './LocationListGridHeader';
import { useNavigate } from 'react-router-dom';
import * as locationListPaths from 'constants/routes';

export const LocationListGrid: React.FC = () => {
  const navigate = useNavigate();

  const handleAddLocation = () => {
    navigate(locationListPaths.newLocation);
  };
  return (
    <>
      <div className="title-block my-3">
        <h2 className="mx-3 my-5">Locations</h2>
      </div>
      <div>
        <div className="row mb-3">
          <div className="col text-right" style={{ minHeight: '40px' }}>
            <button type="submit" className="btn btn-submit d-inline-flex" onClick={handleAddLocation}>
              <span className="icon icon-add" />
              <span className="px-1">Add Location</span>
            </button>
          </div>
        </div>
      </div>
      <div className="container-lg rounded mb-10 ">
        <div className="bg-white table-responsive rounded">
          <table className="table mb-0">
            <LocationListGridHeader />
            <LocationListGridBody />
          </table>
        </div>
      </div>
    </>
  );
};
