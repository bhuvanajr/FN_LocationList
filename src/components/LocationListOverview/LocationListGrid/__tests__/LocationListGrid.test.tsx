import { render, screen } from '@testing-library/react';
import React from 'react';
import { LocationListGrid } from '../LocationListGrid';
import userEvent from '@testing-library/user-event';
import mockSampleData from 'sampleData.json';

const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockNavigate,
}));

jest.mock('context', () => ({
  useLocationListOverviewContext: () => ({
    methods: {
      addNewLocation: (data: Location) => jest.fn(),
      deleteAllLocations: () => jest.fn(),
      deleteLocation: (data: Location) => jest.fn(),
      editLocation: (data: Location) => jest.fn(),
    },
    state: { locationList: mockSampleData, isLoading: false },
  }),
}));

test('should render LocationListGrid correctly', () => {
  const { container } = render(<LocationListGrid />);
  expect(container).toMatchSnapshot();
});

test('should call navigate to add new location route on click of Add location', () => {
  render(<LocationListGrid />);
  userEvent.click(screen.getByText('Add Location'));
  expect(mockNavigate).toHaveBeenCalledWith('/location-list/new');
});
