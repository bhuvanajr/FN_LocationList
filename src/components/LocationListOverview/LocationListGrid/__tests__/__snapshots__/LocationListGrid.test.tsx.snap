// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LocationListGrid correctly 1`] = `
<div>
  <div
    class="title-block my-3"
  >
    <h2
      class="mx-3 my-5"
    >
      Locations
    </h2>
  </div>
  <div>
    <div
      class="row mb-3"
    >
      <div
        class="col text-right"
        style="min-height: 40px;"
      >
        <button
          class="btn btn-submit d-inline-flex"
          type="submit"
        >
          <span
            class="icon icon-add"
          />
          <span
            class="px-1"
          >
            Add Location
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    class="container-lg rounded mb-10 "
  >
    <div
      class="bg-white table-responsive rounded"
    >
      <table
        class="table mb-0"
      >
        <thead
          class="table-light"
        >
          <tr>
            <th
              scope="col"
            >
              Location
            </th>
            <th
              scope="col"
            >
              Location No
            </th>
            <th
              scope="col"
            >
              Chargers
            </th>
            <th
              scope="col"
            >
              Country
            </th>
            <th
              scope="col"
            >
              Last updated
            </th>
            <th
              scope="col"
            />
          </tr>
        </thead>
        <tbody>
          <tr>
            <th
              class="text-primary"
              scope="row"
            >
              Wassenaar
            </th>
            <td>
              345667
            </td>
            <td>
              1
            </td>
            <td>
              NLD
            </td>
            <td>
              8 days ago
            </td>
            <td>
              <button
                class="btn-icon"
                type="button"
              >
                <div
                  class="icon icon-edit"
                />
              </button>
            </td>
          </tr>
          <tr>
            <th
              class="text-primary"
              scope="row"
            >
              Den Haag
            </th>
            <td>
              345667
            </td>
            <td>
              1
            </td>
            <td>
              NLD
            </td>
            <td>
              8 days ago
            </td>
            <td>
              <button
                class="btn-icon"
                type="button"
              >
                <div
                  class="icon icon-edit"
                />
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
